File: Share/src/main/java/com/buguagaoshu/share/domain/Share.java
package com.buguagaoshu.share.domain;

import lombok.Data;

import jakarta.persistence.*;

/**
 * @author Pu Zhiwei {@literal puzhiweipuzhiwei@foxmail.com}
 * create          2020-07-01 20:24
 */
@Data
@Entity
@Table(name = "share_table")
public class Share {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(columnDefinition = "longtext")
    private String data;

    private Long createTime;

    private Long tagId;

    private Boolean lock;


    private String password;

    /**
     * 编辑器类型
     * 默认为 0
     * 0 Vditor https://ld246.com/article/1549638745630
     * 1 wangEditor https://www.wangeditor.com/
     * */
    private Integer editType;

    @Transient
    private Tag tag;
}

File: Go/internal/model/share.go
package model

// Share 对齐 Spring com.buguagaoshu.share.domain.Share（表 share_table）
type Share struct {
	ID         *int64 `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	Data       string `gorm:"column:data;type:longtext" json:"data"`
	CreateTime *int64 `gorm:"column:create_time" json:"createTime"`
	TagId      *int64 `gorm:"column:tag_id" json:"tagId"`
	// Boolean 装箱可空
	Lock       *bool  `gorm:"column:lock" json:"lock"`
	Password   string `gorm:"column:password" json:"password"`
	EditType   *int   `gorm:"column:edit_type" json:"editType"`
	PublicUser *bool  `gorm:"column:public_user" json:"publicUser"`
	Url        string `gorm:"column:url" json:"url"`
	// boolean 基本类型不可空
	HaveUserSeeKey bool `gorm:"column:have_user_see_key;default:false" json:"haveUserSeeKey"`
	UserSeeKey     string `gorm:"column:user_see_key" json:"userSeeKey"`
	CreatePublicShareUserId *int64 `gorm:"column:create_public_share_user_id" json:"createPublicShareUserId"`
	// Tag 对齐 @Transient
	Tag *Tag `gorm:"-" json:"tag"`
}

func (Share) TableName() string { return "share_table" }
